src/modules/todolist.js

```diff
@@ -0,0 +1,138 @@
+  }
+
+  // Retrieve the listArray from Local Storage when the page is loaded..😎😊😉😎............
+
+  // funtion for marking done lists😎😊😉😎............
+
+  listHolder.addEventListener('change', (event) => {
+    if (event.target.type === 'checkbox' && event.target.checked) {
+      event.target.nextElementSibling.style.textDecoration = 'line-through';
+      event.target.completed = true;
+    } else if (event.target.type === 'checkbox' && !event.target.checked) {
+      event.target.nextElementSibling.style.textDecoration = 'none';
+    }
+  });
+
+  if (localStorage.getItem('listArray')) {
+    listArray = JSON.parse(localStorage.getItem('listArray'));
+    UI.displayData();
+  }
+
+  listHolder.addEventListener('click', (event) => {
+    if (event.target.classList.contains('fa-ellipsis-v')) {
+      event.target.classList.remove('fa-ellipsis-v');
+      event.target.classList.add('fa-trash');
+      event.target.parentElement.parentElement.style.backgroundColor = '#f39c12';
+      const targeted = event.target.parentNode.previousElementSibling.lastElementChild;
+      targeted.contentEditable = 'true';
+    } else if (event.target.classList.contains('fa-trash')) {
+    // Get the index of the element to be deleted
+      const trash = event.target;
+      const taskElm = trash.parentNode.parentNode;
+      const listindex = Array.prototype.indexOf.call(listHolder.children, taskElm);
+
+      // Remove the element from the listArray
+      listArray.splice(listindex, 1);
+
+      // Update the local storage with the updated listArray
+      localStorage.setItem('listArray', JSON.stringify(listArray));
+
+      // Remove the element from the DOM
+      event.target.parentElement.parentElement.remove();
+    }
+  });
+
+  listHolder.addEventListener('input', (e) => {
+    if (e.target.closest('#discription')) {
+      const inputText = e.target.closest('#discription');
+      const taskElm = inputText.parentNode.parentNode;
+      const index = Array.prototype.indexOf.call(listHolder.children, taskElm);
+      listArray[index].discription = e.target.textContent;
+
+      localStorage.setItem('listArray', JSON.stringify(listArray));
+    }
+  });
+
+  window.addEventListener('load', () => {
+    if (localStorage.getItem('listArray')) {
+      listArray = JSON.parse(localStorage.getItem('listArray'));
+      UI.displayData();
+    }
+  });
+
+  if (localStorage.getItem('listArray')) {
+    listArray = JSON.parse(localStorage.getItem('listArray'));
+    UI.displayData();
+  }
+
+  // Function to remove completed items from the listArray and the DOM
+  function removeCompleted() {
+    listArray = listArray.filter((item) => !item.completed);
+    localStorage.setItem('listArray', JSON.stringify(listArray));
+    UI.displayData();
+    // Select all elements with the completed class and remove them from the DOM
+    const completedElements = document.querySelectorAll('.completed');
+    completedElements.forEach((element) => element.remove());
+  }
```

Comment:
- [OPTIONAL] As [required ](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md)for this project though not pointed out by the previous code reviewer, " `Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..` ", deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. Please kindly fix that.